tools/tsp-client/src/index.ts

```diff
@@ -0,0 +1,281 @@
+import * as path from "node:path";
+
+import { installDependencies } from "./npm.js";
+import { createTempDirectory, removeDirectory,readTspLocation, getEmitterFromRepoConfig } from "./fs.js";
+import { Logger, printBanner, enableDebug, printVersion } from "./log.js";
+import { compileTsp, discoverMainFile, getEmitterOptions, resolveTspConfigUrl } from "./typespec.js";
+import { getOptions } from "./options.js";
+import { mkdir, writeFile, cp, readFile } from "node:fs/promises";
+import { addSpecFiles, checkoutCommit, cloneRepo, getRepoRoot, sparseCheckout } from "./git.js";
+import { fetch } from "./network.js";
+import { parse as parseYaml } from "yaml";
+
+
+async function sdkInit(
+  {
+    config,
+    outputDir,
+    emitter,
+    commit,
+    repo,
+    isUrl,
+  }: {
+    config: string;
+    outputDir: string;
+    emitter: string;
+    commit: string | undefined;
+    repo: string | undefined;
+    isUrl: boolean;
+  }): Promise<string> {
+  if (isUrl) {
+    // URL scenario
+    const resolvedConfigUrl = await resolveTspConfigUrl(config);
+    Logger.debug(`Resolved config url: ${resolvedConfigUrl.resolvedUrl}`)
+    const tspConfig = await fetch(resolvedConfigUrl.resolvedUrl);
+    const configYaml = parseYaml(tspConfig);
```

Comment:
Can you declare an interface for the expected shape of tspConfig so we don't have to do the heavy `["foo"]` syntax all over the place?

Reply:
Tracking issue to improve in a future PR: https://github.com/Azure/azure-sdk-tools/issues/7047